### PR TITLE
ENH: Adding `order=` keyword to `np.eye()`.

### LINKED ISSUE #9995: Add order='F' option for numpy.eye
For example, in the simplified case that the matrix is square:

```python
id_mat = np.zeros((N, N), order='F')
id_mat.flat[::N + 1] = 1.0
```

I'm happy to send a PR to add this as an extra kwarg, but I don't want to do it unless maintainers are open to it.

It may not be desired to add this. Two reasons come to mind:

- this would add a branch inside `np.eye` for **every** call, which could be a performance hit for code that relies heavily on `np.eye`
- this can already be achieved just by using `np.eye` and then taking the transpose

### PR DESCRIPTION
**Pull request by dhermes**
Fixes #9995.

/cc @eric-wieser @njsmith

### CODE REVIEW

File: numpy/matlib.py [hunk 1]
```diff
@@ -210,7 +213,7 @@ def eye(n,M=None, k=0, dtype=float):
             [ 0.,  0.,  0.]])
 
     """
-    return asmatrix(np.eye(n, M, k, dtype))
+    return asmatrix(np.eye(n, M=M, k=k, dtype=dtype, order=order))
```
Comment (dhermes) [inline numpy/matlib.py]: Switching from positional to keyword is a bit of editorializing by me and I'm happy to revert if desired.

Comment (eric-wieser) [reply numpy/matlib.py]: Keywords are slower, which is why you'll often see them omitted.

File: numpy/tests/test_matlib.py [hunk 2]
```diff
@@ -28,10 +28,19 @@ def test_identity():
...
-                                     [ 0,  0,  0]]))
+    xc = numpy.matlib.eye(3, k=1, dtype=int)
+    assert_array_equal(xc, np.matrix([[ 0,  1,  0],
+                                      [ 0,  0,  1],
+                                      [ 0,  0,  0]]))
+    assert xc.flags.c_contiguous
+    assert not xc.flags.f_contiguous
+
+    xf = numpy.matlib.eye(3, 4, dtype=int, order='F')
+    assert_array_equal(x, np.matrix([[ 1,  0,  0,  0],
```
Comment (eric-wieser) [inline numpy/tests/test_matlib.py]: NameError: global name 'x' is not defined

**Comment by eric-wieser** [comment]
Oh wow, this really is a simple fix!

File: numpy/matlib.py (hunk 1 above)
Comment (eric-wieser) [reply numpy/matlib.py]: On the other hand, the more things to deter users from `matlib`, the better!

File: numpy/lib/twodim_base.py [hunk 3]
```diff
@@ -153,6 +153,9 @@ def eye(N, M=None, k=0, dtype=float):
       to a lower diagonal.
     dtype : data-type, optional
       Data-type of the returned array.
+    order : {'C', 'F'}, optional
+        Whether the output should be stored in row-major (C-style) or
+        column-major (Fortran-style) order in memory.
```
Comment (seberg) [inline numpy/lib/twodim_base.py]: The versionadded :: tag would be good (grep the code for examples).

Comment (seberg) [reply numpy/lib/twodim_base.py]: Ah, also, please format the commit message according to our contribution guide, such as "ENH: ..."

Comment (seberg) [reply numpy/lib/twodim_base.py]: (Then again, I guess nowadays, we can also do that during merging in the squashing, didn't really get used to that yet, heh)

**Comment by charris** [comment]
We should deprecate `matlib` at some point, it is all matrix based and we want to get rid of matrices.

File: numpy/tests/test_matlib.py (hunk 2 above)
Comment (dhermes) [reply numpy/tests/test_matlib.py]: MERP! Copy-pasta (was in a hurry last night and didn't run tests locally, mea culpa).

File: numpy/lib/twodim_base.py (hunk 3 above)
Comment (dhermes) [reply numpy/lib/twodim_base.py]: 1. I said `1.13.4` but feel like that isn't quite the right version?
1. I just read the [commit guide][1], thanks!

[1]: https://docs.scipy.org/doc/numpy-dev/dev/gitwash/development_workflow.html#writing-the-commit-message

**Comment by seberg** [comment]
New features do not go into minor versions, so it will be 1.14 (unless
that would be already frozen, but I don't think it is).

**Comment by dhermes** [comment]
WFM, before I make another commit, would you prefer `1.14` or `1.14.0` in the note? (I assume `1.14` but want to verify.)

**Comment by seberg** [comment]
I am surprised, but it seems 1.14.0 is more common, both is being used
though, so I do no think it matters either way.

**Comment by dhermes** [comment]
@seberg Thanks! Sent a commit with `1.14.0` (also noticed I had accidentally put the `versionadded` on `numpy.matlib.empty`, d'oh!).

**Comment by eric-wieser** [comment]
Thanks @dhermes for spotting this easy fix!

**Pull request merged**